File: 007_boundary/exercise/boundary_length.m
function L = boundary_length(V,F)
%BOUNDARY_LENGTH Compute the length of a mesh's boundary
%
% Input:
%  V,F  input mesh
% Output:
%  L  the length of the boundary

%Compute the indices of all boundary edges.
O = ...

%Compute a matrix whose rows are equal to edgeEnd-edgeStart for each
%boundary edge.
BE = ...

%Compute the total length of all boundary edges.
L = ...

end

File: 007_boundary/exercise/boundary_length.py
import gpytoolbox as gpy, numpy as np

def boundary_length(V,F):
    """Compute the length of the input mesh V,F's boundary.
    """

    # Compute the boundary edges
    bdry_edges = ?

    # Compute a matrix whose rows are equal to edgeEnd-edgeStart for each
    BE = ?

    # Compute the total length of all boundary edges
    # HINT: look up the documentation for the NumPy function np.linalg.norm, and
    #   study the `axis` parameter in particular.
    L = ?

    return L
